Rover/AP_MotorsUGV.cpp

```diff
@@ -613,9 +615,52 @@ void AP_MotorsUGV::output_regular(bool armed, float ground_speed, float steering
         if (_scale_steering) {
             // vectored thrust handling
             if (have_vectored_thrust()) {
-                if (fabsf(throttle) > _vector_throttle_base) {
-                    // scale steering down linearly as throttle increases above _vector_throttle_base
-                    steering *= constrain_float(_vector_throttle_base / fabsf(throttle), 0.0f, 1.0f);
+
+                // normalise desired steering and throttle to ease calculations
+                float steering_norm = steering / 4500.0f;
+                const float throttle_norm = throttle / 1000.0f;
+
+                // steering can never be more than throttle
+                if (fabsf(steering_norm) > fabsf(throttle_norm)) {
```

Comment:
Ah, I've realised that this limitation that, "steering can never be more than throttle" is only correct if the MOT_VEC_ANGLEMAX is 45deg (i.e. the steering can be turned 45deg).  The limitation is actually lower if the angle is less then 45deg and higher if > 45deg.  I'll have to fix this.